modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinBestEffortAffinitySelfTest.java

```diff
@@ -0,0 +1,835 @@
+    /** URL. */
+    private static final String URL = "jdbc:ignite:thin://127.0.0.1:10800..10802?affinityAwareness=true";
+
+    /** Nodes count. */
+    private static final int NODES_CNT = 3;
+
+    /** Query execution multiplier. */
+    private static final int QUERY_EXECUTION_MULTIPLIER = 5;
+
+    /** Rows count. */
+    private static final int ROWS_COUNT = 100;
+
+    /** Connection. */
+    private Connection conn;
+
+    /** Statement. */
+    private Statement stmt;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        CacheConfiguration<?,?> cache = defaultCacheConfiguration();
+
+        cache.setCacheMode(PARTITIONED);
+        cache.setBackups(1);
+        cache.setIndexedTypes(
+            Integer.class, Person.class
+        );
+
+        cfg.setCacheConfiguration(cache);
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        startGridsMultiThreaded(NODES_CNT);
+
+        fillCache(DEFAULT_CACHE_NAME);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        conn = DriverManager.getConnection(URL);
+
+        conn.setSchema('"' + DEFAULT_CACHE_NAME + '"');
+
+        stmt = conn.createStatement();
+
+        assert stmt != null;
+        assert !stmt.isClosed();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        if (stmt != null && !stmt.isClosed()) {
+            stmt.close();
+
+            assert stmt.isClosed();
+        }
+
+        conn.close();
+
+        assert stmt.isClosed();
+        assert conn.isClosed();
+    }
+
+    /**
+     * Check that queries goes to expected number of nodes.
+     *
+     * @throws Exception If failed.
+     */
+    @org.junit.Test
+    public void testExecuteQueries() throws Exception {
+        checkNodesUsage(null, "select * from Person where _key = 1", 1, 1,
+            false);
+
+        checkNodesUsage(null, "select * from Person where _key = 1 or _key = 2", 2,
+            2, false);
+
+        checkNodesUsage(null, "select * from Person where _key in (1, 2)", 2, 2,
+            false);
+    }
+
+    /**
+     * Check that parameterised queries goes to expected number of nodes.
+     *
+     * @throws Exception If failed.
+     */
+    @org.junit.Test
+    public void testExecuteParametrizedQueries() throws Exception {
+        PreparedStatement ps = conn.prepareStatement("select * from Person where _key = ?");
+        ps.setInt(1, 2);
+        checkNodesUsage(ps, null, 1, 1, false);
+
+        ps = conn.prepareStatement("select * from Person where _key = ? or _key = ?");
+        ps.setInt(1, 1);
+        ps.setInt(2, 2);
+        checkNodesUsage(ps, null, 2, 2, false);
```

Comment:
code style - need to add line to separate statements. The same for the whole class.

Reply:
ufff, fixed